pkg/plugins/resources/postgres/plugin.go

```diff
@@ -32,7 +32,14 @@ func (p *plugin) NewResourceStore(pc core_plugins.PluginContext, config core_plu
 	if !migrated {
 		return nil, errors.New(`database is not migrated. Run "kuma-cp migrate up" to update database to the newest schema`)
 	}
-	return NewStore(pc.Metrics(), *cfg)
+	switch cfg.DriverName {
+	case postgres.DriverNamePgx:
+		return NewPgxStore(pc.Metrics(), *cfg)
+	case postgres.DriverNamePq:
+		return NewPqStore(pc.Metrics(), *cfg)
```

Comment:
Technically could wire in the context here with `pc.AppContext()` though I'm not sure what happens when receiving a signal.

Reply:
Is this scope of this PR or should we just create an issue for this? Would love to keep the number of moving parts low.

Reply:
Issue sounds good.

Reply:
https://github.com/kumahq/kuma/issues/6398